pkg/etcd/etcd.go

```diff
@@ -1011,6 +1023,22 @@ func (e *ETCD) RemovePeer(ctx context.Context, name, address string, allowSelfRe
 	return nil
 }
 
+// Creates a specific k8s client for condition patches
+func (e *ETCD) setK8sClient() error {
+	restConfig, err := clientcmd.BuildConfigFromFlags("", e.config.Runtime.KubeConfigSupervisor)
+	if err != nil {
+		return err
+	}
+
+	k8sClient, err := clientk8s.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	e.k8sClient = k8sClient.CoreV1()
+	return nil
```

Comment:
https://github.com/k3s-io/k3s/blob/bbafb86e91ae3682a1811119d136203957df9061/pkg/util/client.go#L30
```suggestion
	e.k8sClient, err := util.GetClientSet(e.config.Runtime.KubeConfigSupervisor)
	return err
```